### PR TITLE
Week 4 - part C - more advanced objects

### PR DESCRIPTION
**Pull request by rarmatei**

### CODE REVIEW

File: week-4/C-more-complex-objects/README.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
-### Goals
+### More objects!
+
+Did you know you can assign arrays to object properties? And even other objects? Or variables?
```
Comment (NateWr) [inline week-4/C-more-complex-objects/README.md]: And even other objects? Or variables? --> Object properties can even be assigned other objects or variables too. The example below shows an object with keys that have been assigned a variable, an array, and an object.

File: week-4/C-more-complex-objects/exercise-1.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// - change the last name of the current owner of "house" to "Montgomery"
+
+
+/*
+  DO NOT EDIT ANYTHING BELOW THIS LINE
+*/
+
+console.log("Expected result: '51 Berkley Road'. Actual result: " + house.address);
+console.log("Expected result: Brian M., Fiona S. Actual result " + house.previousOwners.toString());
+console.log("Expected result: Montgomery. Actual result " + house.currentOwner.lastName);
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-1.js]: Missing `:` after `Actual result` in this line and the one above.

File: week-4/C-more-complex-objects/exercise-1.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*
+  Given the following house - follow the instructions below.
+
+  Make sure you run the file after and it outputs the correct results.
+*/
+
+var house = {
+    address: '1 Kinning Park',
+    previousOwners: ["Claire M.", "John A."],
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-1.js]: In this exercise you mix `"` and `'`. I'm not sure which we're using. Maybe check with Daniel.

File: week-4/C-more-complex-objects/exercise-2.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    currentOwner: {
+      firstName: 'Margaret',
+      lastName: 'Conway'
+    }
+};
+
+var newCurrentOwner = {
+    firstName: 'Margaret',
+    lastName: 'Conway'
+};
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-2.js]: Should this be a different name from `currentOwner`?

File: week-4/C-more-complex-objects/exercise-2.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/*
+    Given the same "house" object again
+
+    Follow the instructions below and then run the file 
+    and make sure it outputs the correct results
+*/
+
+var house = {
+    address: '1 Kinning Park',
+    previousOwners: ["Claire M.", "John A."],
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-2.js]: Mixed `"` and `'` here too.

File: week-4/C-more-complex-objects/exercise-2.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+
+/*
+  DO NOT EDIT ANYTHING BELOW THIS LINE
+*/
+console.log(
+    "Did you correctly assign the new owner using the given variable?",
+    "Expected result: true. Actual result: " + (house.currentOwner === newCurrentOwner));
+console.log("Expected result: Claire M., Stephen B. Actual result " + house.previousOwners.toString());
+console.log("Expected result: false. Actual result " + house.isForSale);
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-2.js]: Missing `:` after `Actual result`

File: week-4/C-more-complex-objects/exercise-3.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    address: '1 Kinning Park',
+    price: 180000,
+    currentOwner: {
+      firstName: 'Margaret',
+      lastName: 'Conway',
+      email: "margaret@fake-emails.com"
+    }
+};
+
+var house = {
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-3.js]: You've defined `house` twice. In the code below you use `house1` and `house2`.

File: week-4/C-more-complex-objects/exercise-3.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    Write the code for the functions as per the description above them
+*/
+
+var house = {
+    address: '1 Kinning Park',
+    price: 180000,
+    currentOwner: {
+      firstName: 'Margaret',
+      lastName: 'Conway',
+      email: "margaret@fake-emails.com"
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-3.js]: Mixed `"` and `'`.

File: week-4/C-more-complex-objects/exercise-3.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+}
+
+
+/*
+  DO NOT EDIT ANYTHING BELOW THIS LINE
+*/
+console.log("Expected result: Margaret Conway. Actual result " + getOwnerFullName(house1));
+console.log("Expected result: margaret@fake-emails.com, marie.m@real-emails.com. Actual result " + getEmailAddresses(house1, house2));
+console.log("Expected result: 1 Kinning Park. Actual result " + getCheapestAddress(house2, house1));
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-3.js]: Missing `:` after `Actual result`.

File: week-4/C-more-complex-objects/exercise-3.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+}
+
+// returns an array of the owners' email addresses of the two houses
+function getEmailAddresses(house1, house2) {
+
+}
+
+// returns the address for the cheapest house out of the two
+function getCheapestAddress(house1, house2) {
```
Comment (NateWr) [inline week-4/C-more-complex-objects/exercise-3.js]: I wonder if it's confusing having global variables `house1` and `house2`, and the same names as arguments in the functions. Perhaps we could use `margaretHouse` and `marieHouse` for the variable names?

File: week-4/C-more-complex-objects/exercise-1.js (hunk 3 above)
Comment (rarmatei) [reply week-4/C-more-complex-objects/exercise-1.js]: I think the the double quotes have been the more popular, and yes, either way, I should keep it consistent and use only one type of quotes

Comment (djgrant) [reply week-4/C-more-complex-objects/exercise-1.js]: Yep. If you have prettier installed and format on save set it should take care of it for you (even in markdown)

File: week-4/C-more-complex-objects/exercise-2.js (hunk 4 above)
Comment (rarmatei) [reply week-4/C-more-complex-objects/exercise-2.js]: Ah yea, it definitely should be :)) Good spot!

File: week-4/C-more-complex-objects/exercise-3.js (hunk 7 above)
Comment (rarmatei) [reply week-4/C-more-complex-objects/exercise-3.js]: good spot!!

File: week-4/C-more-complex-objects/exercise-3.js (hunk 10 above)
Comment (rarmatei) [reply week-4/C-more-complex-objects/exercise-3.js]: Yes! I called them "kinningParkHouse" and "parkAvenueHouse"

**Pull request merged**